packages/wrangler/src/dev.tsx

```diff
@@ -836,11 +836,39 @@ export async function startDev(args: StartDevOptions) {
 
 					logger.log(`${path.basename(config.configPath)} changed...`);
 
+					/*
+					 * If this is a Worker with assets, we want to enable switching
+					 * from assets only Worker to Worker with assets and vice versa
+					 * during local development. For that, we need to watch for changes
+					 * in the `wrangler.toml` file, specifically the `main` configuration
+					 * key, and re-asses the entry point every time.
+					 */
+					if (experimentalAssetsOptions?.directory) {
```

Comment:
technically, we could support this for all Workers, not just Workers with assets. Is this something we would like to do?

Reply:
after internal discussions, the answer is yes. `wrangler dev --x-dev-env` already supports this, no reason why `wrangler dev` shouldn't